src/nunit.analyzers.tests/TestCaseSourceUsage/TestCaseSourceUsesStringAnalyzerTests.cs

```diff
@@ -551,6 +551,109 @@ public class InnerClass
             RoslynAssert.CodeFix(analyzer, fix, expectedDiagnostic.WithMessage(message), testCode, fixedCode);
         }
 
+        [Test]
+        public void NoWarningWhenNumberOfParametersOfTestWillFitIntoParamsArray()
+        {
+            var testCode = TestUtility.WrapClassInNamespaceAndAddUsing(@"
+    [TestFixture]
+    public class NoWarningWhenNumberOfParametersOfTestWillFitIntoParamsArray
+    {
+        [TestCaseSource(↓nameof(TestData))]
+        public void ShortName(params int[] x)
+        {
+            Assert.That(x.Length, Is.GreaterThanOrEqualTo(0));
+        }
+
+        static IEnumerable<int> TestData()
+        {
+            for (int i = 1; i <= 3; i++)
+            {
+                yield return i;
+            }
+        }
+    }", additionalUsings: "using System.Collections.Generic;");
+
+            RoslynAssert.NoAnalyzerDiagnostics(analyzer, testCode);
+        }
+
+        [Test]
+        public void NoWarningWhenNumberOfParametersOfTestWillFitIntoParamsArrayForGenericMethod()
+        {
+            var testCode = TestUtility.WrapClassInNamespaceAndAddUsing(@"
+    [TestFixture]
+    public class NoWarningWhenNumberOfParametersOfTestWillFitIntoParamsArrayForGenericMethod
+    {
+        [TestCaseSource(↓nameof(TestData))]
+        public void ShortName<T>(params T[] x)
+        {
+            Assert.That(x.Length, Is.GreaterThanOrEqualTo(0));
+        }
+
+        static IEnumerable<int> TestData()
+        {
+            for (int i = 1; i <= 3; i++)
+            {
+                yield return i;
+            }
+        }
+    }", additionalUsings: "using System.Collections.Generic;");
```

Comment:
NUnit itself fails to run this test with _Unable to determine type arguments for method_.

I always extract the code from the string to see if NUnit will actually run it.